masonry/src/passes/update.rs

```diff
@@ -145,3 +146,59 @@ pub(crate) fn run_update_pointer_pass(root: &mut RenderRoot, root_state: &mut Wi
     // Pass root widget state to synthetic state create at beginning of pass
     root_state.merge_up(root.widget_arena.get_state_mut(root.root.id()).item);
 }
+
+// ----------------
+
+fn update_disabled_for_widget(
+    global_state: &mut RenderRootState,
+    mut widget: ArenaMut<'_, Box<dyn Widget>>,
+    mut state: ArenaMut<'_, WidgetState>,
+    parent_disabled: bool,
+) {
+    let _span = widget.item.make_trace_span().entered();
+    let id = state.item.id;
+
+    let disabled = state.item.is_explicitly_disabled | parent_disabled;
```

Comment:
I don't like using `BitOr` for bool. This seems an odd thing to try and optimise.

Reply:
Oh yeah you're right, booleans should just use `||`. I actually wonder why it's possible at all in rust (on booleans I mean).

Reply:
Btw. I think we should use the `needless_bitwise_bool` clippy lint?

Reply:
Agreed. Are you happy to PR that?

Reply:
Can do

Reply:
Discussion continues in https://xi.zulipchat.com/#narrow/stream/419691-linebender/topic/Standard.20Lint.20set